### PR TITLE
Unify logging subsystem

### LINKED ISSUE #9: Two different logging subsystems
So we have something in `log.h` and something in `aversive/error.h`. One logging system should be enough for the whole robot I think ;)

I think Aversive's one is more feature rich, we can be both good and bad. In any case it should be cleaned up and maybe simplified (There are definitely some features we don't need).

What do you guys think ? Any reason to keep our own logger ?

I volunteer to cleanup/test/put everywhere Aversive's one should we decide to keep it by the way.

### PR DESCRIPTION
**Pull request by antoinealb**
This PR fixes #9 by moving everything to be managed by Aversive's error handler. There are still a few things I would like to do, including runtime selection of log verbosity and calling functions information that I would like to add. Some cleanup could be useful as well. However, I feel like this work is already useful and therefore can be merged.

@SyrianSpock can you test on hw please ?

### CODE REVIEW

File: master-firmware/tests/error.cpp [hunk 1]
```diff
@@ -0,0 +1,95 @@
+#include <CppUTest/TestHarness.h>
+#include <CppUTestExt/MockSupport.h>
+
+#include "error/error.h"
+
+TEST_GROUP(ErrorLoggingTestGroup)
```
Comment (SyrianSpock) [inline master-firmware/tests/error.cpp]: I would rename `ErrorLoggingTestGroup` to `ErrorLogging` as the "TestGroup" doesn't really adds information and decreases readability.

**Comment by SyrianSpock** [comment]
Works on hardware, sample output from the robot
```
[   0.000000]	/main.c:157	main	UNKNOWN	boot
[   1.115112]	/motor_manager.c:39	main	UNKNOWN	create motor left-wheel
[   1.117028]	/motor_manager.c:39	main	UNKNOWN	create motor right-wheel
```

File: master-firmware/src/debug/log.c [hunk 2]
```diff
@@ -1,32 +1,78 @@
...
+        case ERROR_SEVERITY_ERROR:
+            result = "ERROR";
+        case ERROR_SEVERITY_WARNING:
+            result = "WARNING";
+        case ERROR_SEVERITY_NOTICE:
+            result = "NOTICE";
+        case ERROR_SEVERITY_DEBUG:
+            result = "DEBUG";
+        default:
+            result = "UNKNOWN";
```
Comment (SyrianSpock) [inline master-firmware/src/debug/log.c]: `break` missing, so will always return `UNKNOWN`

Comment (antoinealb) [inline master-firmware/src/debug/log.c]: I guess thats what you get for not unit testing your code... Fixed now.

File: master-firmware/tests/error.cpp (hunk 1 above)
Comment (antoinealb) [inline master-firmware/tests/error.cpp]: Done.

**Comment by antoinealb** [comment]
Pushed some changes which should address your concerns.

File: master-firmware/tests/error.cpp [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    DEBUG("foo");
+}
+
+TEST(ErrorLogging, SeverityName)
+{
+    STRCMP_EQUAL("DEBUG", error_severity_get_name(ERROR_SEVERITY_DEBUG));
+    STRCMP_EQUAL("NOTICE", error_severity_get_name(ERROR_SEVERITY_NOTICE));
+    STRCMP_EQUAL("WARNING", error_severity_get_name(ERROR_SEVERITY_WARNING));
+    STRCMP_EQUAL("ERROR", error_severity_get_name(ERROR_SEVERITY_ERROR));
+}
```
Comment (SyrianSpock) [inline master-firmware/tests/error.cpp]: I would split more

```

TEST_GROUP(ErrorSeverityNameGetter)
{

};

TEST(ErrorSeverityNameGetter, ReturnsErrorStringOnCorrespondingSeverityLevel)
{
    auto severity_name = error_get_severity_name(ERROR_SEVERITY_ERROR);

    STRCMP_EQUAL("ERROR", severity_name);
}

TEST(ErrorSeverityNameGetter, ReturnsWarningStringOnCorrespondingSeverityLevel)
{
    auto severity_name = error_get_severity_name(ERROR_SEVERITY_WARNING);

    STRCMP_EQUAL("WARNING", severity_name);
}

TEST(ErrorSeverityNameGetter, ReturnsNoticeStringOnCorrespondingSeverityLevel)
{
    auto severity_name = error_get_severity_name(ERROR_SEVERITY_NOTICE);

    STRCMP_EQUAL("NOTICE", severity_name);
}

TEST(ErrorSeverityNameGetter, ReturnsDebugStringOnCorrespondingSeverityLevel)
{
    auto severity_name = error_get_severity_name(ERROR_SEVERITY_DEBUG);

    STRCMP_EQUAL("DEBUG", severity_name);
}

TEST(ErrorSeverityNameGetter, ReturnsUnknownStringOnUndefinedSeverityLevel)
{
    const uint8_t undefined_severity_level = 42;
    auto severity_name = error_get_severity_name(undefined_severity_level);

    STRCMP_EQUAL("UNKNOWN", severity_name);
}
```

Comment (SyrianSpock) [inline master-firmware/tests/error.cpp]: We should try to make the tests such that each one fails for a single reason.

Comment (antoinealb) [inline master-firmware/tests/error.cpp]: Fixed.

**Pull request merged**